tealiumlibrary/src/main/java/com/tealium/core/persistence/Serdes.kt

```diff
@@ -0,0 +1,297 @@
+            ?: DoubleArraySerde()).also {
+            doubleArraySerde = it
+            serdeMap.put(DoubleArray::class.java, it)
+        }
+    }
+    fun booleanArraySerde(): Serde<Array<Boolean>> {
+        return (booleanArraySerde
+            ?: BooleanArraySerde()).also {
+            booleanArraySerde = it
+            serdeMap.put(BooleanArray::class.java, it)
+        }
+    }
+    fun jsonObjectSerde(): Serde<JSONObject> {
+        return (jsonObjectSerde
+            ?: JsonObjectSerde()).also {
+            jsonObjectSerde = it
+            serdeMap.put(JSONObject::class.java, it)
+        }
+    }
+    fun jsonArraySerde(): Serde<JSONArray> {
+        return (jsonArraySerde
+            ?: JsonArraySerde()).also {
+            jsonArraySerde = it
+            serdeMap.put(JSONArray::class.java, it)
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    fun <T> serdeFor(clazz: Class<T>) : Serde<T>? {
+        val mapped = serdeMap[clazz]
+        if (mapped != null) {
+            return mapped as Serde<T>
+        }
+
+        return when (clazz) {
+            String::class.java -> stringSerde() as Serde<T>
+            Int::class.java -> intSerde() as Serde<T>
+            Double::class.java -> doubleSerde() as Serde<T>
+            Long::class.java -> longSerde() as Serde<T>
+            Boolean::class.java -> booleanSerde() as Serde<T>
+            Array<String>::class.java -> stringArraySerde() as Serde<T>
+            IntArray::class.java -> intArraySerde() as Serde<T>
+            DoubleArray::class.java -> doubleArraySerde() as Serde<T>
+            LongArray::class.java -> longArraySerde() as Serde<T>
+            BooleanArray::class.java -> booleanArraySerde() as Serde<T>
+            JSONObject::class.java -> jsonObjectSerde() as Serde<T>
+            JSONArray::class.java -> jsonArraySerde() as Serde<T>
+            else -> null
+        }
+    }
+
+    fun <T> serializerFor(clazz: Class<T>) : Serializer<T>? {
+        return serdeFor(clazz)?.serializer
+    }
+
+    fun <T> deserializerFor(clazz: Class<T>) : Deserializer<T>? {
+        return serdeFor(clazz)?.deserializer
+    }
+}
+
+private open class BaseSerde<T>(
+    override val serializer: Serializer<T>,
+    override val deserializer: Deserializer<T>
+) : Serde<T>
+
+private class GenericSerializer<T> : Serializer<T> {
+    override fun serialize(value: T): String {
+        return value.toString()
+    }
+}
+
+private class ArraySerializer<Array> : Serializer<Array> {
+    override fun serialize(value: Array): String {
+        return JSONArray(value).toString()
+    }
+}
+
+private class StringSerde : BaseSerde<String>(GenericSerializer(), StringDeserializer())
+private class StringDeserializer : Deserializer<String> {
+    override fun deserialize(value: String): String {
+        return value
+    }
+}
+
+private class IntSerde : BaseSerde<Int>(GenericSerializer(), IntDeserializer())
+private class IntDeserializer() : Deserializer<Int> {
```

Comment:
I don't think you need the constructor parens here for `IntDeserializer()`, right?